hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataSetLockManager.java

```diff
@@ -256,7 +256,7 @@ private synchronized void putThreadName(String thread) {
     threadCountMap.putIfAbsent(thread, new TrackLog(thread));
   }
 
-  public void lockLeakCheck() {
+  public synchronized void lockLeakCheck() {
```

Comment:
@LiuGuH Thanks for your contribution here. IIRC, this is only invoked when DataNode instance shutdown, and no more concurrent issue generally. So do you meet some issues here? Theoretically, I think it is better to add `synchronized`. Thanks again.

Reply:
Thanks for review.   It is better to add synchronized because  it not , it  is maybe lost some error log.